config/opal_config_hwloc.m4

```diff
@@ -70,6 +70,14 @@ AC_DEFUN([OPAL_CONFIG_HWLOC], [
     AS_IF([test "$external_hwloc_happy" = "0" -a "$internal_hwloc_happy" = "0"],
           [AC_MSG_ERROR([Could not find viable hwloc build.])])
 
+    AS_IF([test "$external_hwloc_happy" = "1"],
```

Comment:
Please use `test "$opal_hwloc_mode" = "external"` here, as that's the proper indicator for an internal vs. external build.